providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/sync/SyncStreamQueueSource.java

```diff
@@ -179,6 +187,7 @@ private void restart() {
             syncRequest.setProviderId(this.providerId);
         }
 
+        incomingQueue.clear();
```

Comment:
When we start a new stream, we should clear our own internal queue. Any unconsumed messages here are for the previous stream and we don't care about them.